src/cdk/stepper/stepper.ts

```diff
@@ -126,6 +156,20 @@ export class CdkStep implements OnChanges {
     return this.stepControl ? this.stepControl.valid && this.interacted : this.interacted;
   }
 
+  /** Whether step has error. */
+  @Input()
+  get hasError(): boolean {
+    return this._customError == null ? this._getDefaultError : this._customError;
+  }
+  set hasError(value: boolean) {
+    this._customError = coerceBooleanProperty(value);
+  }
+  private _customError: boolean | null = null;
+
+  private get _getDefaultError() {
```

Comment:
Getters end up creating alot of extra boilerplate code.  For something like this, can you instead have a private method?

```ts
private _getDefaultError() {
  return this.stepControl && this.stepControl.invalid;
}
```